### PR TITLE
Improve fetch column efficiency use cache, light refactor

### PR DESCRIPTION
**Pull request by LaRiffle**
- Add a method to return sampled columns with improved cache
- Refactor: clean requirements.txt, add .dockerignore, remove the pagai module

### CODE REVIEW

File: pagai/__init__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+from pagai.views import api
+
+app = Flask(__name__)
+app.register_blueprint(api)
+
+
+if __name__ == "__main__":
+    # Load .env config file for entire environement
+    configFileName = (
+        ".env.dev.custom" if os.path.exists(".env.dev.custom") else ".env.dev.default"
```
Comment (simonvadee) [inline pagai/__init__.py]: can you make this configurable ? eg: have an env var like CONFIG_PATH (already talked about this with @elsiehoffet-94 )

File: pagai/engine/main.py [hunk 2]
```diff
@@ -73,7 +70,8 @@ def initialise(self, force_retrain=False):
         # Check model isn't already trained
         if pickle_file.is_file() and not force_retrain:
             logging.warning("Engine already initiated. Loading engine...")
-            engine = pickle.load(pickle_file)
+            with open(pickle_file, "rb") as file:
```
Comment (simonvadee) [inline pagai/engine/main.py]: nit: I think `file` is a reserved keyword in python (like id), could you use something else ? it does not affect the program but it messes with the syntactic coloration

Comment (simonvadee) [reply pagai/engine/main.py]: same below

File: pagai/engine/models/__init__.py [hunk 3]
```diff
@@ -1 +1,3 @@
-from engine.models import train, predict
\ No newline at end of file
+from pagai.engine.models import train, predict
+
+SAVE_PATH = "pickle"
```
Comment (simonvadee) [inline pagai/engine/models/__init__.py]: should this be configurable ?

File: pagai/services/postgres/main.py [hunk 4]
```diff
@@ -117,13 +117,16 @@ def get_table_names(connection=None):
...
 
     for partition in partitions:
-        partition_table_names = partition[0].split(', ')
-        all_table_names = list(filter(lambda x: x[0] not in partition_table_names, all_table_names))
+        partition_table_names = partition[0].split(", ")
+        all_table_names = list(
+            filter(lambda x: x[0] not in partition_table_names, all_table_names)
+        )
 
+    print(np.array(all_table_names))
```
Comment (simonvadee) [inline pagai/services/postgres/main.py]: nit: remove print

File: pagai/services/postgres/main.py [hunk 5]
```diff
@@ -162,35 +165,47 @@ def get_column_names(table, connection=None, include_data_type=False):
...
     table_len = get_length(table, connection)
-    limit = max(5000, round(table_len ** (2 / 3)))
-    query = (
-        "SELECT {} "
-        "FROM {} "
-        "ORDER BY RANDOM() LIMIT {}".format(column, table, limit)
-    )
+    limit = limit or max(5000, round(table_len ** (2 / 3)))
+    order = order or "RANDOM()"
+    query = f"SELECT {column} " f"FROM {table} " f"ORDER BY {order}" f"LIMIT {limit}"
```
Comment (simonvadee) [inline pagai/services/postgres/main.py]: remove extra spaces

File: pagai/services/postgres/main.py [hunk 6]
```diff
@@ -162,35 +165,47 @@ def get_column_names(table, connection=None, include_data_type=False):
...
+storage = {}
+
+
+def get_column_fast(table, column, limit=None, order=None, connection=None):
+    # Build identifier based on args and kwargs
+    identifier = f"{table}:{limit}:{order}"
+
+    # Search storage based on identifier
+    if identifier not in storage:
+        print("compute")
```
Comment (simonvadee) [inline pagai/services/postgres/main.py]: nit: remove print

File: pagai/services/postgres/main.py [hunk 7]
```diff
@@ -162,35 +165,47 @@ def get_column_names(table, connection=None, include_data_type=False):
...
+
+        columns = get_column_names(table, connection)
+
+        results = np.array(results)
+
+        storage[identifier] = {}
+        for idx, col in enumerate(columns):
+            storage[identifier][col] = results[:, idx]
+    else:
+        print("cached")
```
Comment (simonvadee) [inline pagai/services/postgres/main.py]: remove this `else` (or check that column is in storage[identifier])

File: pagai/services/postgres/main.py [hunk 8]
```diff
@@ -203,33 +218,37 @@ def fetch_columns(datasets, dataset_size, connection=None):
         datasets = [datasets]
 
     columns = []
-    for dataset in tqdm(datasets):
+    print(len(datasets), "datasets")
```
Comment (simonvadee) [inline pagai/services/postgres/main.py]: nit: print

File: pagai/__init__.py (hunk 1 above)
Comment (elsiehoffet-94) [reply pagai/__init__.py]: For example: `configFileName = os.getenv("CONFIG_PATH",'.env.dev.default')`

File: pagai/engine/models/__init__.py (hunk 3 above)
Comment (LaRiffle) [reply pagai/engine/models/__init__.py]: I don't think so, it's internal and shouldn't be acceed by the suer

**Pull request merged**